lib/matplotlib/collections.py

```diff
@@ -899,6 +900,118 @@ def set_paths(self, paths):
     def get_paths(self):
         return self._paths
 
+    def legend_items(self, mode="colors", useall="auto", num=10,
+                     fmt=None, func=lambda x: x, **kwargs):
+        """
+        Creates legend handles and labels for a PathCollection. This is useful
+        for obtaining a legend for a :meth:`~.Axes.scatter` plot. E.g.::
+
+            scatter = plt.scatter([1,2,3], [4,5,6], c=[7,2,3])
+            plt.legend(*scatter.legend_items())
```

Comment:
I'm not sure I like the name `legend_items` just because it makes it sound like these are a property of the PathCollection;  I think a verb would help like `create_legend_items` or `make_legend_items`

Reply:
For consistency I would go with `legend_elements` as commented above already. Unless people are prepared to break the existing contour method and call *both* something else?

Reply:
I still like having a verb because you are actually doing something versus just returning a property of the collection.  But I don't feel super strongly...